src/genotyper.cc

```diff
@@ -413,13 +391,50 @@ Status genotype_site(const genotyper_config& cfg, MetadataCache& cache, BCFData&
             }
         }
 
-        // for each source BCF record
-        for (const auto& record : records) {
-            S(translate_genotypes(cfg, site, dataset, dataset_header.get(), record.get(),
+        // TODO: Lift update_orig_calls_for_loss to here instead of translate_genotype for accurate accounting
+
+        if (records.size() > 1) {
+            // Multiple source bcf overlapping with site
+            // We need to "join" these records
+
+            // We first find the number of non-gvcf entries in these records
+            vector<shared_ptr<bcf1_t>> non_gvcf_records;
+            copy_if(records.begin(), records.end(), back_inserter(non_gvcf_records), [&cfg](auto& record) {return !is_gvcf_ref_record(cfg, record.get()); });
+
+            if (non_gvcf_records.size() == 0) {
+                // this region is all gvcf records, we can interpret it as REF
+                // by passing it any gvcf REF record
+                S(translate_genotypes(cfg, site, dataset, dataset_header.get(), records[0].get(),
+                                  sample_mapping, genotypes, genotyped, loss_trackers));
+            } else if (non_gvcf_records.size() == 1) {
+                // this region contains a single variant record and >=1 gvcf
+                // confidence records. We can use the variant record to accurately
+                // perform translate_genotype due to setup from unifier
+                S(translate_genotypes(cfg, site, dataset, dataset_header.get(), non_gvcf_records[0].get(),
+                                  sample_mapping, genotypes, genotyped, loss_trackers));
+            } else {
+                // Multiple variant records in this range.
```

Comment:
I'm planning to lift this code path off completely (ie we don't bother with `translate_genotype` when it's a case that we can't resolve). It is left here for the moment to complete loss tracking.